Index: src/cbit/vcell/message/server/htc/pbs/PbsProxy.java
===================================================================
--- src/cbit/vcell/message/server/htc/pbs/PbsProxy.java	(revision 12075)
+++ src/cbit/vcell/message/server/htc/pbs/PbsProxy.java	(revision 12074)
@@ -8,7 +8,6 @@
 import java.io.StringReader;
 import java.io.StringWriter;
 import java.util.ArrayList;
-import java.util.Comparator;
 import java.util.List;
 import java.util.StringTokenizer;
 import java.util.TreeMap;
@@ -302,13 +301,7 @@
 		try {
 			String[] cmd = new String[]{JOB_CMD_STATUS, "|", "grep", serverID.toString().toUpperCase()+"_"};
 			CommandOutput commandOutput = commandService.command(cmd);
-			TreeMap<HtcJobID, String> pbsJobIDMapServiceType =
-				new TreeMap<HtcJobID, String>(new Comparator<HtcJobID>() {
-					@Override
-					public int compare(HtcJobID o1, HtcJobID o2) {
-						return Integer.parseInt(((PbsJobID)o2).getPbsJobID())-Integer.parseInt(((PbsJobID)o1).getPbsJobID());
-					}
-				});
+			TreeMap<HtcJobID, String> pbsJobIDMapServiceType = new TreeMap<HtcJobID, String>();
 			BufferedReader br = new BufferedReader(new StringReader(commandOutput.getStandardOutput()));
 			String line = null;
 			while((line = br.readLine()) != null){
Index: src/cbit/vcell/message/server/htc/sge/SgeProxy.java
===================================================================
--- src/cbit/vcell/message/server/htc/sge/SgeProxy.java	(revision 12075)
+++ src/cbit/vcell/message/server/htc/sge/SgeProxy.java	(revision 12074)
@@ -6,7 +6,6 @@
 import java.io.StringReader;
 import java.io.StringWriter;
 import java.util.ArrayList;
-import java.util.Comparator;
 import java.util.HashMap;
 import java.util.List;
 import java.util.StringTokenizer;
@@ -396,13 +395,7 @@
 		try{
 			String[] cmd = new String[]{JOB_CMD_STATUS, "-xml"};//get running jobs in XML format
 			CommandOutput commandOutput = commandService.command(cmd);
-			TreeMap<HtcJobID, String> pbsJobIDMapServiceType =
-				new TreeMap<HtcJobID, String>(new Comparator<HtcJobID>() {
-					@Override
-					public int compare(HtcJobID o1, HtcJobID o2) {
-						return Integer.parseInt(((SgeJobID)o2).getSgeJobID())-Integer.parseInt(((SgeJobID)o1).getSgeJobID());
-					}
-				});
+			TreeMap<HtcJobID, String> pbsJobIDMapServiceType = new TreeMap<HtcJobID, String>();
 			Document qstatDoc = XmlUtil.stringToXML(commandOutput.getStandardOutput(), null);
 			Element rootElement = qstatDoc.getRootElement();
 			List<Element> qstatInfoChildren = rootElement.getChild("queue_info").getChildren("job_list");
