Index: src/cbit/vcell/message/server/sim/HtcSimulationWorker.java
===================================================================
--- src/cbit/vcell/message/server/sim/HtcSimulationWorker.java	(revision 12094)
+++ src/cbit/vcell/message/server/sim/HtcSimulationWorker.java	(revision 12093)
@@ -126,7 +126,7 @@
 		// compiled solver ...used to be only single executable, now we pass 2 commands to PBSUtils.submitJob that invokes SolverPreprocessor.main() and then the native executable
 		String[] preprocessorCmd = new String[] { 
 				PropertyLoader.getRequiredProperty(PropertyLoader.simulationPreprocessor), 
-//				serviceInstanceStatus.getServerID().toString().toLowerCase(), 
+				serviceInstanceStatus.getServerID().toString().toLowerCase(), 
 				simTaskFilePath, 
 				forceUnixPath(userdir.getAbsolutePath())
 		};
