Index: src/cbit/vcell/message/server/htc/HtcProxy.java
===================================================================
--- src/cbit/vcell/message/server/htc/HtcProxy.java	(revision 12265)
+++ src/cbit/vcell/message/server/htc/HtcProxy.java	(revision 12264)
@@ -14,14 +14,11 @@
 import java.util.TreeMap;
 import java.util.Vector;
 
-import org.vcell.util.BeanUtils;
 import org.vcell.util.ExecutableException;
 import org.vcell.util.document.KeyValue;
 import org.vcell.util.document.VCellServerID;
 
 import cbit.vcell.message.server.cmd.CommandService;
-import cbit.vcell.message.server.cmd.CommandServiceLocal;
-import cbit.vcell.message.server.cmd.CommandServiceSsh;
 import cbit.vcell.message.server.htc.pbs.PbsProxy;
 
 public abstract class HtcProxy {
@@ -181,17 +178,4 @@
 	public abstract String getSubmissionFileExtension();
 
 	public abstract void checkServerStatus() throws ExecutableException;
-	
-	/**
-	 * for unix-style commands that have (e.g. pipes), using the java Runtime.exec(String[] cmd), the pipe requires a shell to operate properly.
-	 * the SSH command already invokes this with a shell, so it is not required.
-	 */
-	protected final String[] constructShellCommand(CommandService commandService, String[] cmd){
-		if (commandService instanceof CommandServiceSsh){
-			return cmd;
-		}else if (commandService instanceof CommandServiceLocal){
-			return BeanUtils.addElements(new String[] { "/bin/sh","-c"}, cmd);
-		}
-		throw new RuntimeException("expected either SSH or Local CommandService");
-	}
 }
Index: src/cbit/vcell/message/server/htc/pbs/PbsProxy.java
===================================================================
--- src/cbit/vcell/message/server/htc/pbs/PbsProxy.java	(revision 12265)
+++ src/cbit/vcell/message/server/htc/pbs/PbsProxy.java	(revision 12264)
@@ -329,7 +329,7 @@
 	@Override
 	public List<HtcJobID> getRunningJobIDs(String jobNamePrefix) throws ExecutableException {
 		try {
-			String[] cmd = constructShellCommand(commandService, new String[]{JOB_CMD_STATUS, "|", "grep", jobNamePrefix});
+			String[] cmd = new String[]{JOB_CMD_STATUS, "|", "grep", jobNamePrefix};
 			CommandOutput commandOutput = commandService.command(cmd);
 			ArrayList<HtcJobID> pbsJobIDs = new ArrayList<HtcJobID>();
 			BufferedReader br = new BufferedReader(new StringReader(commandOutput.getStandardOutput()));
@@ -402,7 +402,7 @@
 
 		try{
 			
-			String[] commandArray = constructShellCommand(commandService,new String[]{QSTAT_FULL_CLUSTER_COMMAND_PATH,"|", "grep "+HTC_SIMULATION_JOB_NAME_PREFIX});
+			String[] commandArray = new String[]{QSTAT_FULL_CLUSTER_COMMAND_PATH,"|", "grep "+HTC_SIMULATION_JOB_NAME_PREFIX};
 			CommandOutput commandOutput = commandService.command(commandArray);
 			if (commandOutput.getExitStatus()==1) {return null;} //because Grep returns code 1 if nothing found
 			if (commandOutput.getExitStatus()!=0 || commandOutput.getStandardOutput()==null) {
Index: src/cbit/vcell/message/server/htc/sge/SgeProxy.java
===================================================================
--- src/cbit/vcell/message/server/htc/sge/SgeProxy.java	(revision 12265)
+++ src/cbit/vcell/message/server/htc/sge/SgeProxy.java	(revision 12264)
@@ -419,7 +419,7 @@
    6951 0.55500 TEST2_MySe vcell        r     10/10/2012 19:08:34 all.q@compute-4-1.local            1
    6952 0.55500 TEST2_MySe vcell        r     10/10/2012 19:08:34 all.q@compute-0-1.local            1
  */
-		String[] cmd = constructShellCommand(commandService, new String[]{JOB_CMD_STATUS, "|", "grep", jobNamePrefix});
+		String[] cmd = new String[]{JOB_CMD_STATUS, "|", "grep", jobNamePrefix};
 		CommandOutput commandOutput = commandService.command(cmd);
 		ArrayList<HtcJobID> serviceJobIDs = new ArrayList<HtcJobID>();
 		
