Index: src/cbit/vcell/message/server/sim/HtcSimulationWorker.java
===================================================================
--- src/cbit/vcell/message/server/sim/HtcSimulationWorker.java	(revision 12096)
+++ src/cbit/vcell/message/server/sim/HtcSimulationWorker.java	(revision 12095)
@@ -147,7 +147,7 @@
 		File inputFile = new File(getBaseName(userdir,simTask)+SimDataConstants.JAVA_INPUT_EXTENSION);
 		String[] command = new String[] { 
 				PropertyLoader.getRequiredProperty(PropertyLoader.javaSimulationExecutable), 
-//				VCellServerID.getSystemServerID().toString(), 
+				VCellServerID.getSystemServerID().toString(), 
 				forceUnixPath(inputFile.getParent()), 
 				forceUnixPath(inputFile.getName()), 
 				String.valueOf(simTask.getSimulationJob().getJobIndex()),
