Index: src/cbit/vcell/message/server/manager/ServerManagerDaemon.java
===================================================================
--- src/cbit/vcell/message/server/manager/ServerManagerDaemon.java	(revision 12072)
+++ src/cbit/vcell/message/server/manager/ServerManagerDaemon.java	(revision 12071)
@@ -308,7 +308,6 @@
 			System.setErr(ps);			
 		}
 
-		org.vcell.util.PropertyLoader.loadProperties();
 		CommandService commandService = null;
 		if (args.length==4){
 			String htcHost = args[1];
@@ -327,6 +326,7 @@
 		}else{
 			throw new RuntimeException("unsupported batch system "+batchSystemType);
 		}
+		org.vcell.util.PropertyLoader.loadProperties();
 		ServiceInstanceStatus serviceInstanceStatus = new ServiceInstanceStatus(VCellServerID.getSystemServerID(), ServiceType.SERVERMANAGER, 0, ManageUtils.getHostName(), new Date(), true); 
 		SessionLog log = new StdoutSessionLog(serviceInstanceStatus.getID());
 		ConnectionFactory conFactory = null;
