diff --git a/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java b/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java
index e09cfd088..c399a8460 100644
--- a/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java
+++ b/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java
@@ -12,6 +12,7 @@
 
 package org.vcell.restclient.api;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import org.vcell.restclient.ApiClient;
 import org.vcell.restclient.ApiException;
 import org.vcell.restclient.ApiResponse;
@@ -276,16 +277,20 @@ public class FieldDataResourceApi {
     multiPartBuilder.addBinaryBody("file", _file);
     hasFiles = true;
     multiPartBuilder.addTextBody("fileName", fileName.toString());
-    multiPartBuilder.addTextBody("extent", extent.toString());
-    multiPartBuilder.addTextBody("iSize", iSize.toString());
-    for (int i=0; i < channelNames.size(); i++) {
-        multiPartBuilder.addTextBody("channelNames", channelNames.get(i).toString());
+    try {
+      multiPartBuilder.addTextBody("extent", memberVarObjectMapper.writeValueAsString(extent));
+      multiPartBuilder.addTextBody("iSize", memberVarObjectMapper.writeValueAsString(iSize));
+      multiPartBuilder.addTextBody("origin", memberVarObjectMapper.writeValueAsString(origin));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
     }
     for (int i=0; i < times.size(); i++) {
-        multiPartBuilder.addTextBody("times", times.get(i).toString());
+      multiPartBuilder.addTextBody("times", times.get(i).toString());
+    }
+    for (int i=0; i < channelNames.size(); i++) {
+      multiPartBuilder.addTextBody("channelNames", channelNames.get(i).toString());
     }
     multiPartBuilder.addTextBody("annotation", annotation.toString());
-    multiPartBuilder.addTextBody("origin", origin.toString());
     HttpEntity entity = multiPartBuilder.build();
     HttpRequest.BodyPublisher formDataPublisher;
     if (hasFiles) {
