diff --git a/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java b/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java
index d32ccaf1cd..1337f17437 100644
--- a/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java
+++ b/vcell-restclient/src/main/java/org/vcell/restclient/api/FieldDataResourceApi.java
@@ -12,6 +12,7 @@
 
 package org.vcell.restclient.api;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import org.vcell.restclient.ApiClient;
 import org.vcell.restclient.ApiException;
 import org.vcell.restclient.ApiResponse;
@@ -316,11 +317,18 @@ public class FieldDataResourceApi {
     if (fileName != null) {
         multiPartBuilder.addTextBody("fileName", fileName.toString());
     }
-    if (extent != null) {
-        multiPartBuilder.addTextBody("extent", extent.toString());
-    }
-    if (iSize != null) {
-        multiPartBuilder.addTextBody("iSize", iSize.toString());
+    try {
+        if (extent != null) {
+            multiPartBuilder.addTextBody("extent", memberVarObjectMapper.writeValueAsString(extent));
+        }
+        if (iSize != null) {
+            multiPartBuilder.addTextBody("iSize", memberVarObjectMapper.writeValueAsString(iSize));
+        }
+        if (origin != null) {
+            multiPartBuilder.addTextBody("origin", memberVarObjectMapper.writeValueAsString(origin));
+        }
+    } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
     }
     for (int i=0; i < channelNames.size(); i++) {
         if (channelNames.get(i) != null) {
@@ -335,9 +343,6 @@ public class FieldDataResourceApi {
     if (annotation != null) {
         multiPartBuilder.addTextBody("annotation", annotation.toString());
     }
-    if (origin != null) {
-        multiPartBuilder.addTextBody("origin", origin.toString());
-    }
     HttpEntity entity = multiPartBuilder.build();
     Supplier<InputStream> formDataSupplier;
     if (hasFiles) {
